### DISCUSSION TITLE
How to swap from external to build in Syncthing?

### DISCUSSION THREAD

**Discussion by marsch1975** [Martchus/syncthingtray#307] [General]
Hello everyone,

I changed the Syncthing tray version from 1.5.? to 1.6.4 and the Syncthing version from 1.27.? to 1.28.1.

After that Syncthing didn't shut down anymore when Windows wanted to shut down, in other words, it stopped Windows from shutting down. Does anyone have the same problem and how can it be solved?

So I was thinking maybe I should swap to the internal Syncthing version in Syncthing tray.

I ticket the box "use internal Syncthing-library". Then I was asked for the "configurations folder" and "data folder". What does that mean? Is it the folder where the former Syncthing has all the configuration in? Like this:

c:\Users\Markus\AppData\Local\Syncthing\

And what does "replace $[var]..." mean?

**Comment by Martchus** [comment]
> After that Syncthing didn't shut down anymore when Windows wanted to shut down, in other words, it stopped Windows from shutting down. Does anyone have the same problem and how can it be solved?

Not sure as I only use the stand-alone/official Syncthing executable for testing on Windows.

> So I was thinking maybe I should swap to the internal Syncthing version in Syncthing tray.

That should definitely help with the problem because Syncthing Tray will shut down and also shut down Syncthing itself when started via its launcher. It would only block Windows from shutting down if that's not happening in a timely manner for some reason.

> I ticket the box "use internal Syncthing-library". Then I was asked for the "configurations folder" and "data folder". What does that mean? Is it the folder where the former Syncthing has all the configuration in? Like this:

Ticking the checkbox is enough and leave everything else empty. (The further configurations are only interesting if you want to divert from the default paths. The internal Syncthing library uses the same default paths as the standalone executable.)

> c:\Users\Markus\AppData\Local\Syncthing\

I guess that's just the default so yes, really just leave the settings for the paths empty.

I suppose I should add a placeholder text like "leave empty to use default path".

> And what does "replace $[var]..." mean?

This is a feature another user asked for. So I don't use it myself but when I remember correctly then it replaces these placeholders/variables with values from corresponding environment variable. You probably don't need that.

**Comment by marsch1975** [comment]
Hi Martchus,

thanks for you help!

I left the paths empty and it worked out. Yes, a placeholder would have been great.

AND MOST IMPORTANT.
SyncthingTray can now close the Syncthing instance and Windows can shut down agein!!! :)

All the best
Markus

**Reply by Martchus** [reply]
> I left the paths empty and it worked out. Yes, a placeholder would have been great.

I added a placeholder text yesterday.

> AND MOST IMPORTANT.
> SyncthingTray can now close the Syncthing instance and Windows can shut down agein!!! :)

How did you start Syncthing before? If you used the Syncthing Tray launcher to start an external executable shutting down should arguably not have been a problem as well. That is because Syncthing Tray is supposed to stop Syncthing via a REST-API call and offer to terminate it forcefully (via `TerminateProcess`) in case it doesn't shutdown in a timely mannder. (Under non-Windows systems it simply sends SIGTERM instead of using the REST-API but under Windows that's not possible.) Hard to tell why this didn't work in your case.

**Comment by marsch1975** [comment]
I started Syncthing with SyncthingTray as well but the external Version of Syncthing. It was working up to Version 1.27.7 but now with 1.28.1 it wasn't anymore... On all computers...

Now with starting the internal Syncthing version of SyncthingTray 1.6.4 it's fine on all computers...

the Computers are running Win 11 23H2 and Win 10...
